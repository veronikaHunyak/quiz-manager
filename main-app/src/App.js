import React, { Component } from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from "react-router-dom";

import Login from "./components/login";

export const userTypes = {
  edit: "Edit", //with username of user1
  view: "View", //with username of user2
  restricted: "Restricted", //with username of user3
};

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      userType: localStorage.getItem("user"),
      loggedIn: false,
    };
  }

  componentDidMount() {
    if (this.state.userType) {
      this.setState({ loggedIn: true });
    } else {
      this.setState({ loggedIn: false });
    }
  }

  logOut = () => {
    this.setState({ loggedIn: false });
    localStorage.removeItem("user");
  };

  loginStateChange = (loginType) => {
    localStorage.setItem("user", loginType);
    this.setState({ loggedIn: true, userType: loginType });
  };

  renderlogin() {
    if (!this.state.loggedIn) {
      return (
        <Link to="/login" className="nav-link">
          Login
        </Link>
      );
    } else {
      return (
        <Link to="/" className="nav-link" onClick={this.logOut}>
          Logout
        </Link>
      );
    }
  }

  render() {
    return (
      <Router>
        <div>
          <nav
            className="navbar navbar-expand-lg navbar-light"
            style={{ backgroundColor: "aliceblue" }}
          >
            <Link to="/quizes" className="navbar-brand">
              Quiz Manager
            </Link>
            <button
              className="navbar-toggler"
              type="button"
              data-toggle="collapse"
              data-target="#navbarText"
              aria-controls="navbarText"
              aria-expanded="false"
              aria-label="Toggle navigation"
              style={{ backgroundColor: "whitesmoke" }}
            >
              <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id="navbarText">
              <ul className="navbar-nav mr-auto">
                <li className="navbar-item">
                  <Link to="/quizes" className="nav-link">
                    Quizes
                  </Link>
                </li>
                <li className="navbar-item">{this.renderlogin()}</li>
              </ul>
              {this.state.loggedIn && (
                <p>
                  Welcome back, your access rights are:{" "}
                  <b>{this.state.userType}</b>
                </p>
              )}
            </div>
          </nav>

          <Switch>
            <Route
              path="/login"
              exact
              render={(props) => (
                <Login {...props} loginStateChange={this.loginStateChange} />
              )}
            >
              {this.state.loggedIn ? (
                <Redirect to="/quizes" />
              ) : (
                <Login loginStateChange={this.loginStateChange} />
              )}
            </Route>

            {/* <Route
              path="/quizes"
              render={(props) => (
                <ViewQuizes
                  {...props}
                  loggedInUserType={this.state.userType}
                  quizes={this.state.quizes}
                />
              )}
            /> */}

            <Route path="/" exact>
              {this.state.loggedIn ? (
                <Redirect to="/quizes" />
              ) : (
                <Login loginStateChange={this.loginStateChange} />
              )}
            </Route>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
