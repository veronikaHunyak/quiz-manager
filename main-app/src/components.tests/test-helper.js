export const mockQuiz = [
  {
    _id: "5eb14db7c8248d0f7cd4a279",
    quizKey: "quiz0",
    quizName: "First quiz test name",
    quizData: [
      {
        question: "Question 1",
        options: [{ A: "one" }, { B: "two" }, { C: "three" }],
        answer: "B",
      },
      {
        question: "question2",
        options: [{ A: "a" }, { B: "b" }, { C: "c" }, { D: "d" }],
        answer: "C",
      },
    ],
  },
];
