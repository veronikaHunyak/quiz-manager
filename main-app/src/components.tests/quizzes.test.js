import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import ViewQuizzes from "../components/quizzes";
import { userTypes } from "../App";
import { mockQuiz } from "./test-helper";

describe("ViewQuizzes", () => {
  const props = {
    loggedInUserType: userTypes.edit,
    quizzes: mockQuiz,
  };

  it("should render correctly", () => {
    expect(toJson(shallow(<ViewQuizzes {...props} />))).toMatchSnapshot();
  });

  it("should match snapshot of the user has view rights", () => {
    const wrapper = shallow(
      <ViewQuizzes {...props} loggedInUserType={userTypes.view} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot of the user has restricted rights", () => {
    const wrapper = shallow(
      <ViewQuizzes {...props} loggedInUserType={userTypes.restricted} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
