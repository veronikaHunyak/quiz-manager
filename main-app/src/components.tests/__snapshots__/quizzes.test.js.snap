// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewQuizzes should match snapshot of the user has restricted rights 1`] = `
<div>
  <a
    href="/add-quizzes"
  />
  <br />
  <div
    className="container-fluid"
  >
    <div
      className="list-group"
    >
      <a
        href="/questions/5eb14db7c8248d0f7cd4a279"
        key="0"
      >
        <button
          className="list-group-item list-group-item-action list-group-item-info"
          style={
            Object {
              "margin": "5px",
            }
          }
          type="button"
        >
          First quiz test name
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`ViewQuizzes should match snapshot of the user has view rights 1`] = `
<div>
  <a
    href="/add-quizzes"
  />
  <br />
  <div
    className="container-fluid"
  >
    <div
      className="list-group"
    >
      <a
        href="/questions/5eb14db7c8248d0f7cd4a279"
        key="0"
      >
        <button
          className="list-group-item list-group-item-action list-group-item-info"
          style={
            Object {
              "margin": "5px",
            }
          }
          type="button"
        >
          First quiz test name
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`ViewQuizzes should render correctly 1`] = `
<div>
  <a
    href="/add-quizzes"
  >
    <button
      className="btn btn-danger"
      style={
        Object {
          "margin": "10px",
        }
      }
      type="button"
    >
      <PlusSquare
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      Add a new quiz
    </button>
  </a>
  <br />
  <div
    className="container-fluid"
  >
    <div
      className="list-group"
    >
      <a
        href="/questions/5eb14db7c8248d0f7cd4a279"
        key="0"
      >
        <button
          className="list-group-item list-group-item-action list-group-item-info"
          style={
            Object {
              "margin": "5px",
            }
          }
          type="button"
        >
          First quiz test name
        </button>
      </a>
    </div>
  </div>
</div>
`;
